lang/java/avro/src/main/java/org/apache/avro/io/DatumWriter.java

```diff
@@ -36,4 +36,11 @@ public interface DatumWriter<D> {
    * the schema from the datum to the output.
    */
   void write(D datum, Encoder out) throws IOException;
+
+  /**
+   * Convenience method to Write a datum to a byte array. Traverse the schema,
+   * depth first, writing each leaf value in the schema from the datum to the byte
+   * array.
+   */
+  byte[] toByteArray(D datum) throws IOException;
```

Comment:
The two implementations are the same at the moment. Would it make sense to make this method a method with default implementation?

Reply:
Touché. Much Cleaner. Thanks for the suggestion. Updated.